contrib/src/main/java/com/datatorrent/contrib/parquet/ParquetFilePOJOReader.java

```diff
@@ -0,0 +1,305 @@
+  protected static final String FIELD_SEPARATOR = ":";
+  protected static final String RECORD_SEPARATOR = ",";
+
+  @OutputPortFieldAnnotation(schemaRequired = true)
+  public final transient DefaultOutputPort<Object> output = new DefaultOutputPort<Object>()
+  {
+
+    @Override
+    public void setup(Context.PortContext context)
+    {
+      pojoClass = context.getValue(Context.PortContext.TUPLE_CLASS);
+      pojoSetters = Maps.newHashMap();
+      for (Field f : pojoClass.getDeclaredFields()) {
+        try {
+          pojoSetters.put(f.getName(), generateSettersForField(pojoClass, f.getName()));
+        } catch (NoSuchFieldException | SecurityException e) {
+          e.printStackTrace();
+        }
+      }
+
+      if (groupToPOJOFieldsMapping == null) {
+        fieldInfos = createFieldInfoMap(generateFieldInfoInputs());
+      } else {
+        fieldInfos = createFieldInfoMap(groupToPOJOFieldsMapping);
+      }
+      initColumnFieldSetters();
+    }
+
+  };
+
+  /**
+   * Converts Group to POJO
+   */
+  @Override
+  protected Object convertGroup(Group group)
+  {
+    Object obj;
+    try {
+      obj = pojoClass.newInstance();
+    } catch (InstantiationException | IllegalAccessException ex) {
+      throw new RuntimeException(ex);
+    }
+    for (int i = 0; i < columnFieldSetters.size(); i++) {
+      ParquetFilePOJOReader.ActiveFieldInfo afi = columnFieldSetters.get(i);
+
+      afi = columnFieldSetters.get(i);
+      int fieldIndex = schema.getFieldIndex(afi.fieldInfo.getColumnName());
+      SupportType st = afi.fieldInfo.getType();
+
+      switch (st) {
+
+        case BOOLEAN:
+          Boolean booleanVal = Boolean.parseBoolean(group.getValueToString(fieldIndex, 0));
+          pojoSetters.get(afi.fieldInfo.getPojoFieldExpression()).set(obj, booleanVal);
+          break;
+
+        case INTEGER:
+          Integer intVal = Integer.parseInt(group.getValueToString(fieldIndex, 0));
+          pojoSetters.get(afi.fieldInfo.getPojoFieldExpression()).set(obj, intVal);
+          break;
+
+        case LONG:
+          Long longVal = Long.parseLong(group.getValueToString(fieldIndex, 0));
+          pojoSetters.get(afi.fieldInfo.getPojoFieldExpression()).set(obj, longVal);
+          break;
+
+        case FLOAT:
+          Float floatVal = Float.parseFloat(group.getValueToString(fieldIndex, 0));
+          pojoSetters.get(afi.fieldInfo.getPojoFieldExpression()).set(obj, floatVal);
+          break;
+
+        case DOUBLE:
+          Double doubleVal = Double.parseDouble(group.getValueToString(fieldIndex, 0));
+          pojoSetters.get(afi.fieldInfo.getPojoFieldExpression()).set(obj, doubleVal);
+          break;
+
+        case STRING:
+          pojoSetters.get(afi.fieldInfo.getPojoFieldExpression()).set(obj, group.getValueToString(fieldIndex, 0));
+          break;
+
+        default:
+          throw new ParquetEncodingException("Unsupported column type: " + schema.getType(fieldIndex));
```

Comment:
Can the default case be to serialize the record as a byte array and set it on the object ? This could help serializing non-primitive types as byte arrays, which the user can then deserialize.